src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts

```diff
@@ -74,6 +74,28 @@ export const hasUncheckedASharedType = (
   return false;
 };
 
+const hasPendingTypeChange = (
+  type: ClientType,
+  isChecked: boolean,
+  sharedTypes: ClientType[]
+): boolean =>
+  (!isChecked && sharedTypes.includes(type)) || (isChecked && !sharedTypes.includes(type));
```

Comment:
maybe just `! (isChecked === sharedTypes.includes(type))`?

Reply:
haha why didn't I think of that - done!